EIPS/eip-7685.md

```diff
@@ -0,0 +1,76 @@
+---
+eip: 7685
+title: General purpose execution layer requests
+description: A general purpose bus for sharing EL triggered requests with the CL
+author: lightclient (@lightclient)
+discussions-to: https://ethereum-magicians.org/t/eip-7685-general-purpose-execution-layer-requests/19668
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-04-14
+---
+
+## Abstract
+
+This proposal defines a general purpose framework for propagating contract-triggered requests from execution layer to the consensus layer. It extends the header and body with only a single field each to store the request information.
+
+## Motivation
+
+The proliferation of smart contract controlled validators has caused there to be a demand for additional EL triggered behaviors. By allowing these systems to delegate administrative operations to their governing smart contracts, they can avoid intermediaries needing to step in and ensure certain operations occur. This creates a safer system for end users.
+
+## Specification
+
+### Execution Layer
+
+#### Request
+
+A `Request` consists of a `RequestType` prepended to an opaque byte array `RequestData`.
+
+```
+Request = RequestType ++ RequestData
+```
+
+Let `Requests` be the list of all `Request` objects in the block in ascending order by type.
+
+#### Header
+
+Extend the header with a new 32 byte value `RequestsRoot`.
```

Comment:
Can this be expanded to provide the complete structure of the RLP encoding of the header, along the lines of [4788](https://github.com/ethereum/EIPs/blob/6a0a792731c16d92d56e0525dd0abff7b498d7b7/EIPS/eip-4788.md?plain=1#L51)

On a related note, given we're adding this new field, does this imply that 4844 and 4788 are required EIPs for this in the sense that they have both added new header fields that are prior to `requestsRoot` in the header field order?